flink-clients/src/main/java/org/apache/flink/client/cli/ExecutionParameterProviderBuilder.java

```diff
@@ -0,0 +1,161 @@
+import static org.apache.flink.client.cli.CliFrontendParser.CLASS_OPTION;
+import static org.apache.flink.client.cli.CliFrontendParser.DETACHED_OPTION;
+import static org.apache.flink.client.cli.CliFrontendParser.JAR_OPTION;
+import static org.apache.flink.client.cli.CliFrontendParser.PARALLELISM_OPTION;
+import static org.apache.flink.client.cli.CliFrontendParser.PYMODULE_OPTION;
+import static org.apache.flink.client.cli.CliFrontendParser.PY_OPTION;
+import static org.apache.flink.client.cli.CliFrontendParser.SAVEPOINT_ALLOW_NON_RESTORED_OPTION;
+import static org.apache.flink.client.cli.CliFrontendParser.SAVEPOINT_PATH_OPTION;
+import static org.apache.flink.client.cli.CliFrontendParser.SHUTDOWN_IF_ATTACHED_OPTION;
+import static org.apache.flink.client.cli.CliFrontendParser.YARN_DETACHED_OPTION;
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Creates a {@link ExecutionParameterProvider} based either on a {@link Configuration}, or a {@link CommandLine command-line options}.
+ */
+@Internal
+public class ExecutionParameterProviderBuilder {
+
+	/**
+	 * Creates a {@link ExecutionParameterProvider} based on the provided {@link Configuration}.
+	 */
+	public static ExecutionParameterProvider fromConfiguration(final Configuration configuration) {
+		return new ExecutionParameterProvider(checkNotNull(configuration));
+	}
+
+	/**
+	 * Creates a {@link ExecutionParameterProvider} based on the provided user-provided {@link CommandLine command-line options}.
+	 */
+	public static ExecutionParameterProvider fromCommandLine(final CommandLine commandLine) throws CliArgsException {
+		final Configuration configuration = new Configuration();
+
+		final String[] args = commandLine.hasOption(ARGS_OPTION.getOpt()) ?
+				commandLine.getOptionValues(ARGS_OPTION.getOpt()) :
+				commandLine.getArgs();
+
+		final String entryPointClass = commandLine.hasOption(CLASS_OPTION.getOpt())
+				? commandLine.getOptionValue(CLASS_OPTION.getOpt())
+				: null;
+
+		final boolean isPython = commandLine.hasOption(PY_OPTION.getOpt())
+				| commandLine.hasOption(PYMODULE_OPTION.getOpt())
+				| "org.apache.flink.client.python.PythonGatewayServer".equals(entryPointClass);
+
+		if (commandLine.hasOption(JAR_OPTION.getOpt())) {
+			parseJarURLToConfig(commandLine.getOptionValue(JAR_OPTION.getOpt()), configuration);
+		} else if (!isPython && args.length > 0) {
+			parseJarURLToConfig(args[0], configuration);
+		}
+
+		parseClasspathURLsToConfig(commandLine, configuration);
+		parseParallelismToConfig(commandLine, configuration);
+		parseDetachedModeToConfig(commandLine, configuration);
+		parseShutdownOnExitToConfig(commandLine, configuration);
+		parseSavepointRestoreSettingsToConfig(commandLine, configuration);
+
+		return new ExecutionParameterProvider(configuration);
+	}
+
+	private static boolean parseShutdownOnExitToConfig(final CommandLine commandLine, final Configuration configuration) {
```

Comment:
Anywhere we need return values of these `parse` functions? According to `parse` prefix I would assume it is action only.